### PR TITLE
Revert "fix(pkgs/flox): use semver build metadata"

### PR DESCRIPTION
**Pull request by garbas**
Reverting https://github.com/flox/flox/pull/2212

### CODE REVIEW

File: pkgs/flox/default.nix [hunk 1]
```diff
@@ -20,9 +20,9 @@ let
     if (FLOX_VERSION != null) then
       FLOX_VERSION
     else if !(self ? revCount || self ? shortRev) then
-      "${fileVersion}+git.dirty"
+      "${fileVersion}-dirty"
     else if !(self ? revCount) then
-      "${fileVersion}+git.${self.shortRev}"
+      "${fileVersion}-g${self.shortRev}"
```
Comment (mkenigs) [inline pkgs/flox/default.nix:25]: Maybe we should add some comments so we know where we depend on this and why we use this format? So we don't break it again in the future

Comment (ysndr) [reply pkgs/flox/default.nix:25]: yes a _reason_ would be great, even though i know its probably https://github.com/semver/semver/issues/394

**Comment by ysndr** [comment]
@garbas looks like you got the git hooks disabled or click through the gh interface, but since we're rebasing onto main on merge, could you adjust the commit messages and rebase rather than merge main?

**Comment by garbas** [comment]
@ysndr don't know the rules you have in this repo. Feel free to change the commits however you need them to, I've squashed the commits.

**Comment by ysndr** [comment]
yeah i figured, non worries, thanks
I opened #2224 to have the commit messages checked in CI as well

**Review by bryanhonof — APPROVED** [review]

**Pull request merged**